libshaderc_util/src/compiler.cc

```diff
@@ -67,6 +67,45 @@ std::pair<int, string_piece> DecodeLineDirective(string_piece directive) {
   return std::make_pair(line, directive);
 }
 
+// Writes the message contained in the diagnostic parameter to *dest. Assumes
+// the diagnostic message is reported for for a binary location.
+void OutputSpvToolsDiagnostic(spv_diagnostic diagnostic, std::string* dest) {
+  assert(!diagnostic->isTextSource);
+
+  std::ostringstream os;
+  os << diagnostic->position.index << ": " << diagnostic->error;
+  *dest = os.str();
+}
+
+// Disassembles the given binary. Returns SPV_SUCCESS and writes the
+// disassembled text to *text_or_error if successful. Otherwise, writes the
+// error message to *text_or_error.
+spv_result_t DisassembleBinary(const std::vector<uint32_t>& binary,
+                               std::string* text_or_error) {
+  auto spvtools_context = spvContextCreate();
```

Comment:
The SPIRV-Tools context could be made a member of the Compiler object.  Then we wouldn't have to create and destroy the SPIR-V Tools context object for each disassembled module.

Right now it does an allocation, but long term, I'd like to precompile all the data in the SPIR-V Tools context object so it wouldn't need extra memory or time to set up.  
But that hasn't happened just yet, so I'd feel better pushing this context object out to the Compiler object.